fabric-resource-conditions-api-v1/src/main/java/net/fabricmc/fabric/api/resource/conditions/v1/ConditionJsonProvider.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.resource.conditions.v1;
+
+import com.google.common.base.Preconditions;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.util.Identifier;
+
+/**
+ * A resource condition and its parameters that can be serialized to JSON, meant for use in data generators.
+ */
+public interface ConditionJsonProvider {
+	/**
+	 * Write the passed conditions to a JSON object in the {@value ResourceConditions#CONDITIONS_KEY} array.
+	 *
+	 * @throws IllegalArgumentException if the JSON object already contains that array
+	 */
+	static void write(JsonObject conditionalObject, ConditionJsonProvider @Nullable... conditions) {
+		if (conditions == null) { // no condition -> skip
```

Comment:
It'd be better if we handle the null and the 0-length array case the same

Reply:
The 0-length case is usually a programmer error, while the `null` case is conveniently used to skip unspecified entries in the datagen.

Reply:
> The 0-length case is usually a programmer error, while the `null` case is conveniently used to skip unspecified entries in the datagen.

Then why not just ask the modders to not call this method if they have nothing to write to the conditions after all

Reply:
Well I guess modders could make use of it in their own datagen if they need their own conditions